packages/Python/lldbsuite/test/lang/swift/swift_interface_shared/TestSwiftInterfaceNoDebugInfo.py

```diff
@@ -0,0 +1,135 @@
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# -----------------------------------------------------------------------------
+"""
+Test that we load and handle modules that only have textual .swiftinterface
+files -- i.e. no associated .swiftmodule file -- and no debug info. The module
+loader should generate the .swiftmodule for any .swiftinterface it finds unless
+it is already in the module cache.
+"""
+
+import commands
+import glob
+import lldb
+from lldbsuite.test.lldbtest import *
+import lldbsuite.test.decorators as decorators
+import lldbsuite.test.lldbutil as lldbutil
+import os
+import os.path
+import unittest2
+
+
+class TestSwiftInterfaceNoDebugInfo(TestBase):
+    mydir = TestBase.compute_mydir(__file__)
+
+    @decorators.swiftTest
+    def test_swift_interface(self):
+        """Test that we load and handle modules that only have textual .swiftinterface files"""
+        self.build()
+        self.do_test()
+
+
+    def setUp(self):
+        TestBase.setUp(self)
+        self.main_source = "main.swift"
+        self.main_source_spec = lldb.SBFileSpec(self.main_source)
+
+
+    def do_test(self):
+        # The custom module cache location
+        mod_cache = self.getBuildArtifact("MCP")
+
+        # Clear the module cache (populated by the Makefile build)
+        shutil.rmtree(mod_cache)
+        self.assertFalse(os.path.isdir(mod_cache),
+                         "module cache should not exist")
+
+        # Update the settings to use the custom module cache location
+        self.runCmd('settings set symbols.clang-modules-cache-path "%s"' % mod_cache)
+
+        # Set a breakpoint in and launch the main executable
+        (target, process, thread, bkpt) = lldbutil.run_to_source_breakpoint(self,
+                "break here", self.main_source_spec,
+                exe_name="main")
+
+        self.frame = thread.frames[0]
+        self.assertTrue(self.frame, "Frame 0 is valid.")
+
+        # Check we are able to access the public fields of variables whose
+        # types are from the .swiftinterface-only dylibs
+        var = self.frame.FindVariable("x")
+        lldbutil.check_variable(self, var, False, typename="AA.MyPoint")
+
+        child_y = var.GetChildMemberWithName("y") # MyPoint.y is public
+        lldbutil.check_variable(self, child_y, False, value="0")
+
+        child_x = var.GetChildMemberWithName("x") # MyPoint.x isn't public
+        self.assertFalse(child_x.IsValid())
+
+        # Expression evaluation using types from the .swiftinterface only
+        # dylibs should work too
+        lldbutil.check_expression(self, self.frame, "y.magnitudeSquared", "404", use_summary=False)
+        lldbutil.check_expression(self, self.frame, "MyPoint(x: 1, y: 2).magnitudeSquared", "5", use_summary=False)
+
+        # Check the module cache was populated with the .swiftmodule files of
```

Comment:
Can you spell this "Swift module cache" to avoid confusing with the clang module cache?